XunitTestingForApi/AddPostUnitTest.cs

```diff
@@ -0,0 +1,177 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using postapi.Controllers;
+using postapi.Model;
+using postapi.Repository;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using Xunit;
+
+namespace XunitTestingForApi
+{
+    public class AddPostUnitTest1
+    {
+
+        [Fact]
+        public void GetPosts_Returns_All_posts()
+        {
+
+            // Arrange
+            var mockRepo = new Mock<IAddPostRepository>();
+
+            mockRepo.Setup(repo => repo.GetPosts())
+            .Returns(GetTestPost());
+
+            var controller = new AddPostController(mockRepo.Object);
+
+            var result = controller.GetPosts();
+            var objectResult = (ObjectResult)result;
+
+
+            Assert.NotNull(result);
+            Assert.Equal(StatusCodes.Status200OK, objectResult.StatusCode);
+            Assert.Equal(HttpStatusCode.OK, (HttpStatusCode)objectResult.StatusCode);
+            Assert.Collection((List<Post>)objectResult.Value,
+                item =>
+                {
+                    Assert.Equal(1, item.PostID);
+                    Assert.Equal("Test post", item.PostTittle);
+                    Assert.Equal("Test desc", item.DescriptionOfPost);
+                    Assert.Equal("Test desc", item.DescriptionOfPost);
+                    Assert.Equal(Convert.ToDateTime("22-01-2021"), item.CreatedDate);
+                    Assert.Equal(Convert.ToDateTime("23-01-2021"), item.UpdatedDate);
+                    Assert.Equal(1, item.LikesCount);
+                    Assert.Equal(1, item.HeartCount);
+                });
+        }
+
+
+        [Fact]
+        public void CreatePost_Returns_Post()
+        {
+
+
+            // Arrange
+            var mockRepo = new Mock<IAddPostRepository>();
+
+            var post = new Post()
+            {
+                PostTittle = "Test post",
+                DescriptionOfPost = "description"
+            };
+
+            mockRepo.Setup(repo => repo.CreatePost(post));
+
+            var controller = new AddPostController(mockRepo.Object);
+
+            var result = controller.CreatePost(post);
+            var objectResult = (ObjectResult)result;
+
+
+            Assert.NotNull(result);
+            Assert.Equal(StatusCodes.Status200OK, objectResult.StatusCode);
+            Assert.Equal(HttpStatusCode.OK, (HttpStatusCode)objectResult.StatusCode);
+
+
+
+        }
+
+        [Fact]
+        public void UpdatePost_Returns_Post()
+        {
+
+
```

Comment:
Can you please format this file and format CommentsUnitTest.cs file as well, It should not have extra blank lines/spaces.